# Problema 1

MATRICES

Orden de datos en filas y columnas, siempre se definen con mayuscula y los elementos que la componen con su letra minuscula justo al numero de fila y columna.

A = [a11, a12, a13
     a21, a22, a23]

El orden de una matriz es mencionar cuantas filas y columnas tiene.

Orden A2x3

i -> fila
j -> columna

cuando hay que llamar columnas y filas en python siempre hay que restarle uno, si es en la fila 3, llamas a la fila 2(lo mismo con las columnas).

Las matrices $A$ y $B$ representan las ventas en dos años, en millones de dólares, de los tres productos de una empresa en las cuatro regiones en que está disponible. Es decir, $a_{ij}$ y $b_{ij}$ representan el monto de las ventas del producto $i$ en la región $j$ el primer y segundo año, respectivamente:

<br>

\begin{split}
A = \begin{pmatrix}
 2,6 &  4,8 &  1,8 & 0,9 \\
 3,2 &  4,4 &  2,5 & 2,8 \\
 2,4 &  3,6 &  3,8 & 2,5 \\
\end{pmatrix} \text{ } B= \begin{pmatrix}
 3,6 &  2,5 &  3,0 & 2,5 \\
 4,5 &  5,0 &  3,5 & 3,8 \\
 2,9 &  3,0 &  4,6 & 4,0 \\
\end{pmatrix}
\end{split}

<br>

1. Indique el orden de cada matriz.
2. Calcule la matriz $S = A + B$.
3. Calcule la matriz $D = B - A$.
4. Indique el valor e interprete los elementos $s_{24}$ y $d_{24}$.
5. Para el tercer año de funcionamiento, los tres productos ahora están disponibles en seis regiones. Durante este año, las ventas de esta empresa estarán registradas en una matriz $E$. ¿Es posible determinar la matriz $T = A + B + E$? Justifique su respuesta.

In [9]:
print('R1: El orden en ambas matrices es 3x4.')

import numpy as np

A = np.array([[2.6, 4.8, 1.8, 0.9],
              [3.2, 4.4, 2.5, 2.8],
              [2.4, 3.6, 3.8, 2.5]])

B = np.array([[3.6, 2.5, 3, 2.5],
              [4.5, 5, 3.5, 3.8],
              [2.9, 3, 4.6, 4]])

S = A + B
print(f'R2: S={S}')

D =B-A
print(f'R3: D={D}')

print(f'R4: s24= {S[1][3]}; El total de ventas del producto 2 en la region 4 contando los dos años es de 6.6 millones de dolares.')
print(f'    d24={D[1][3]}; En el segundo año las ventas del producto 2 en la region 4 aumentaron en un 1 millon de dolares.')

print('R5: No es posible determinar T porque la matriz E tiene un distinto orden (3x6).')

R1: El orden en ambas matrices es 3x4.
R2: S=[[6.2 7.3 4.8 3.4]
 [7.7 9.4 6.  6.6]
 [5.3 6.6 8.4 6.5]]
R3: D=[[ 1.  -2.3  1.2  1.6]
 [ 1.3  0.6  1.   1. ]
 [ 0.5 -0.6  0.8  1.5]]
R4: s24= 6.6; El total de ventas del producto 2 en la region 4 contando los dos años es de 6.6 millones de dolares.
    d24=1.0; En el segundo año las ventas del producto 2 en la region 4 aumentaron en un 1 millon de dolares.
R5: 


# Problema 2

Una fábrica con sucursales en Japón y China, produce tres tipos de componentes $j$, para cinco modelos de computador $i$. Los costos de producción mensual, en miles de dólares, de ambas sucursales se representan en las matrices $J$ y $C$, respectivamente:

<br>

\begin{split}
J = \begin{pmatrix}
 100 &  200 &  150  \\
 120 &  210 &  170  \\
 112 &  200 &  131  \\
 124 &  310 &  170  \\
 136 &  156 &  140  \\
\end{pmatrix} \text{ } C = \begin{pmatrix}
 110 &  190 &  160  \\
 130 &  220 &  180  \\
 146 &  190 &  141  \\
 143 &  312 &  163  \\
 137 &  116 &  156  \\
\end{pmatrix}
\end{split}

<br>

1. ¿Cuál es el orden de las matrices?
2. Inteprete el elemento $j_{21}$ y $c_{12}$ en el contexto del problema.
3. Utilizando *Python*, calcule el costo total de producción para cada modelo, considerando ambas sucursales.
4. Un trabajador indica que, en la sucursal japonesa, el costo de producción del producto $3$, modelo $4$ de computador, es más alto que en la sucursal china. ¿Tiene razón? Justifique apropiadamente.

In [21]:
print('R1: El orden en ambas matrices es 5x3.')

import numpy as np

J = np.array([[100, 200, 150],
              [120, 210, 170],
              [112, 200, 131],
              [124, 310, 170],
              [136, 156, 140]])

C = np.array([[110, 190, 160],
              [130, 220, 180],
              [146, 190, 141],
              [143, 312, 163],
              [137, 116, 156]])
print('R2:')
print(f'En la sucursal de Japon, el costo de produccion mensual del primer componente para el segundo modelo del computador es de {J[1][0]} miles de dolares.')
print(f'En la sucursal de China, el costo de produccion mensual, del segundo componente para el primer modelo del computador es de {C[0][1]} miles de dolares')

T = J + C
print(f'R3: {T}')

print(f'R4: El costo de produccion del producto 3, modelo 4 en Japon: {J[3][2]} miles de dolares, China: {C[3][2]} miles de dolares. El trabajador tiene razon ya que el costo es mal alto en la sucursal de Japon.')




R1: El orden en ambas matrices es 5x3.
R2:
En la sucursal de Japon, el costo de produccion mensual del primer componente para el segundo modelo del computador es de 120 miles de dolares.
En la sucursal de China, el costo de produccion mensual, del segundo componente para el primer modelo del computador es de 190 miles de dolares
R3: [[210 390 310]
 [250 430 350]
 [258 390 272]
 [267 622 333]
 [273 272 296]]
R4: El costo de produccion del producto 3, modelo 4 en Japon: 170 miles de dolares, China: 163 miles de dolares. El trabajador tiene razon ya que el costo es mal alto en la sucursal de Japon.


# Problema 3

La empresa Beta tiene cinco plantas en Chile, y en cada una de ellas fabrica siete productos. El número de unidades del producto $i$ fabricadas en la planta $j$ en una semana está representado en la matriz $C$ en el elemento $c_{ij}$.

<br>

\begin{split}
C = \begin{pmatrix}
 120 &  200 &  155 &  125  &  230 \\
 155 &  130 &  110 &  160  &  180 \\
 140 &  205 &  150 &  125  &  155 \\
 130 &  120 &  230 &  225  &  125 \\
 135 &  70  &  190 &  175  &  95  \\
 110 &  265 &  180 &  230  &  180 \\
 185 &  85  &  120 &  135  &  150 \\
\end{pmatrix}
\end{split}

<br>

La empresa se expande, creando cinco plantas en Ecuador que producen los mismos siete productos. La producción semanal de esta filial estará representada por una matriz $E$.

1. Si se sabe que los volúmenes de producción semanal de la filial en Ecuador son un $20\%$ menor que en Chile, calcule la matriz $E$ utilizando *Python*.
2. Si representamos la producción semanal total de la empresa Beta, considerando ambas filiales, por la matriz $T$, escriba la expresión que calcula cada elemento de la matriz $T$ usando notación matricial.
3. Calcule la matriz $T$ utilizando *Python*.
4. Interprete los elementos $c_{34}$ y $t_{72}$ en el contexto del problema.

In [34]:
C = np.array([[120, 200, 155, 125, 230],
              [155, 130, 110, 160, 180],
              [140, 205, 150, 125, 155],
              [130, 120, 230, 225, 125],
              [135, 70, 190, 175, 95],
              [110, 265, 180, 230, 180],
              [185, 85, 120, 135, 150]])

E= C*0.8
print('R1:')
print(f'E: {E}')

T = C+E

print(f'R2 y R3: La Matriz T es:\n {T}')

print('')
print(f'R4: En la fabrica 4 de Chile se fabricaron del productos 3: {C[2][3]} unidades.')
print(f' El total de unidades del producto 7 que se fabricaron en las fabricas 2 de Chile y Ecuador es de {T[6][1]:.0f} unidades.')

R1:
E: [[ 96. 160. 124. 100. 184.]
 [124. 104.  88. 128. 144.]
 [112. 164. 120. 100. 124.]
 [104.  96. 184. 180. 100.]
 [108.  56. 152. 140.  76.]
 [ 88. 212. 144. 184. 144.]
 [148.  68.  96. 108. 120.]]
R2 y R3: La Matriz T es:
 [[216. 360. 279. 225. 414.]
 [279. 234. 198. 288. 324.]
 [252. 369. 270. 225. 279.]
 [234. 216. 414. 405. 225.]
 [243. 126. 342. 315. 171.]
 [198. 477. 324. 414. 324.]
 [333. 153. 216. 243. 270.]]

R4: En la fabrica 4 de Chile se fabricaron del productos 3: 125 unidades.
 El total de unidades del producto 7 que se fabricaron en las fabricas 2 de Chile y Ecuador es de 153 unidades.


# Problema 4

Una empresa que fabrica zapatos tiene dos plantas, una en Valparaíso y la otra en Temuco. La empresa produce zapatos de color negro, blanco y café, tanto para niños, damas como caballeros. La capacidad actual de producción mensual (en miles de pares) para Valparaíso y Temuco, están dadas por las matrices $V$ y $T$, respectivamente. Para ambas matrices, según el orden en que se mencionan, las filas corresponden al color del calzado y las columnas al tipo de persona que utilizará el calzado.

<br>

\begin{split}
V = \begin{pmatrix}
 30 &  34 &  20  \\
 45 &  20 &  16  \\
 14 &  26 &  25  \\
\end{pmatrix} \text{ } T = \begin{pmatrix}
 35 &  30 &  26  \\
 52 &  25 &  18  \\
 23 &  24 &  32  \\
\end{pmatrix}
\end{split}

<br>

1. Determine la producción mensual total de ambas plantas, representándola en una matriz $A$. Utilice *Python* para resolver.
2. Interprete el elemento $a_{21}$ en el contexto del problema.
3. Si para el próximo año se estima que la producción en Valparaíso aumenta en un $30\%$, mientras que la de Temuco se reduce en un $10\%$, represente la nueva producción total en la matriz $B$.
4. Interprete el elemento $b_{13}$ en el contexto del problema.

In [22]:
import numpy as np

V =np.array([[30, 34, 20],
             [45, 20, 16],
             [14, 26, 25]])

T =np.array([[35, 30, 26],
             [52, 25, 18],
             [23, 24, 32]])

A = V + T

print(f'R1: Matriz A: \n{A}')
print('')
print(f'Las ambas plantas de la empresa fabrica mensualmente {A[1][0]} miles de pares de zapatos blancos para niños.')
print('')

B = 1.3*V + 0.9*T

print(f' Matriz B: \n{B}')
print('')

print(f'Se estima que el proximo año,ambas plantas fabricaran {B[0][2]:.1f} miles de pares de zapatos negros para caballeros.')

R1: Matriz A: 
[[65 64 46]
 [97 45 34]
 [37 50 57]]

Las ambas plantas de la empresa fabrica mensualmente 97 miles de pares de zapatos blancos para niños.

 Matriz B: 
[[ 70.5  71.2  49.4]
 [105.3  48.5  37. ]
 [ 38.9  55.4  61.3]]

Se estima que el proximo año,ambas plantas fabricaran 49.4 miles de pares de zapatos negros para caballeros. 


# Problema 5

Genere una matriz $M$ de orden $8 \times 9$ donde cada término de la matriz está definido por:

$$m_{ij} = 8i + 12j$$

y calcule su transpuesta.

# Problema 6

Genere dos matrices:

*   Una matriz $A$ de orden $5 \times 7$, con $a_{ij} = i \cdot j.$
*   Una matriz $B$ de orden $7 \times 5$, con $b_{ij} = 15i - 2j.$

y calcule $C = A^{T} + \dfrac{1}{2}B$.



# Problema 7

Una red de sensores recoge datos de temperatura ($°C$), humedad (porcentaje) y presión (hPa) en cinco días distintos (de lunes a viernes). Los datos medidos se muestran en la matriz $D$:

<br>

\begin{equation*}
\begin{matrix}
& \begin{matrix}
 & Lu  \ \ &Ma \ \ &Mi \ \ & Ju \ \ &Vi
\end{matrix} \\
&D = \begin{pmatrix}
20    &  28    &  22    &  26     &  24   \\
52    &  57    &  56    &  54     &  58   \\
1.066 &  1.068 &  1.066 &  1.064  &  1.068 \\
\end{pmatrix} & \ \begin{matrix}
Cº\\
\% \\
hPa
\end{matrix}\\
\end{matrix}
\end{equation*}

<br>

Se sabe que por un error de calibración de los sensores, las medidas registradas deberían ser un $5\%$ menores que las registradas.

1. Calcule en una matriz $S$ los valores correctos de temperatura, humedad y presión. Utilice *Python* para calcular.
2. Obtenga la matriz $S^{T}$.
3. Interprete los elementos $s_{24}$ y $s_{31}^{T}$.


# Problema 8

Una empresa tecnológica vende software y hardware en tres productos: $P_{1}$, $P_{2}$ y $P_{3}$, en cuatro zonas de la capital: norte, sur, este y oeste. Las ventas del mes, en millones de pesos, están registradas en las matrices $S$ y $H$, para software y hardware respectivamente.

<br>

\begin{split}
S = \begin{pmatrix}
 260 &  480 &  180  &  90  \\
 320 &  440 &  250  &  280 \\
 240 &  360 &  380  &  250  \\
\end{pmatrix} \text{ } H = \begin{pmatrix}
 360 &  250 &  300  &  250  \\
 450 &  500 &  350  &  380 \\
 290 &  300 &  460  &  400  \\
\end{pmatrix}
\end{split}

<br>

1. Calcule en la matriz $T$, del total de ventas de la empresa tecnológica.
2. Un trabajador indica que este mes, la sucursal de la zona sur recaudó más por la venta hardware que software, ¿es correcta su afirmación?
3. Debido al black friday se espera que el siguiente mes, las ventas de software aumenten un $7\%$ y las de hardware aumenten un $3\%$ respecto al mes actual. Encuentra en total de ventas esperadas para el siguiente mes y almacénalo en la matriz $M$. Utiliza *Python* para realizar los cálculos.
4. Interpreta los elementos $t_{24}$ y $m_{31}$ en el contexto del problema.
5. Calcula la matriz $M^{T}$ e interpreta el elemento $m_{41}^{T}$ en el contexto del problema.
